agent/consul/state/catalog.go

```diff
@@ -103,6 +105,47 @@ func gatewayServicesTableNameSchema() *memdb.TableSchema {
 	}
 }
 
+// topologyTableNameSchema returns a new table schema used to store information
+// relating upstream and downstream services
+func topologyTableNameSchema() *memdb.TableSchema {
```

Comment:
doesn't this need to be split into oss/ent schemas so that the namespace indexes are handled properly?

Reply:
It won't need that. `ServiceNameIndex` calls `ServiceName.String()` when computing an index, and that `String()` function has the oss/ent split.

Reply:
Ah I guess you only need split indexes if you explicitly want to query by _just_ a namespace for something.